File: typescript/src/utils/common/valueHelpers.ts
import { Value } from '../../types/blocks';

export function literalString(value?: Value): string | undefined {
    if (value?.type === 'literal' && typeof value.value === 'string') {
        return value.value;
    }
    return undefined;
}

export function literalBoolean(value?: Value): boolean | undefined {
    if (value?.type === 'literal' && typeof value.value === 'boolean') {
        return value.value;
    }
    return undefined;
}

export function literalNumber(value?: Value): number | undefined {
    if (value?.type === 'literal' && typeof value.value === 'number') {
        return value.value;
    }
    return undefined;
}

File: python/src/parse_hcl/value_helpers.py
from __future__ import annotations

from typing import Any, Optional

from .types import Value


def literal_string(value: Value | None) -> Optional[str]:
    if value and value.get("type") == "literal" and isinstance(value.get("value"), str):
        return value["value"]  # type: ignore[return-value]
    return None


def literal_boolean(value: Value | None) -> Optional[bool]:
    if value and value.get("type") == "literal" and isinstance(value.get("value"), bool):
        return value["value"]  # type: ignore[return-value]
    return None


def literal_number(value: Value | None) -> Optional[float]:
    if value and value.get("type") == "literal" and isinstance(value.get("value"), (int, float)):
        return float(value["value"])  # type: ignore[return-value]
    return None
